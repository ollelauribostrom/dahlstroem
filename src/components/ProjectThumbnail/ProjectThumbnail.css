.project__thumbnail {
  position: relative;
  margin: 20px;
}

.project__thumbnail__image {
  width: 260px;
  height: 330px;
  justify-self: center;
}

.project__thumbnail__title {
  position: absolute;
  bottom: 0px;
  left: 0px;
  width: 100%;
  text-align: center;
  padding: 10px 0;
  background-color: rgba(0, 0, 0, 0.5);
  color: #fff;
}

@media only screen and (max-device-width: 1024px) {
  .project__thumbnail__image {
    width: 200px;
    height: 250px;
    justify-self: center;
  }
}
