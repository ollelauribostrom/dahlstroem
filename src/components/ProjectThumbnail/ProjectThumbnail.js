import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'gatsby';
import Img from 'gatsby-image';

class ProjectThumbnail extends React.Component {
  constructor(props) {
    super(props);
    this.state = { showTitle: false };
  }
  render() {
    const { slug, featuredImage, title } = this.props.project;
    return (
      <Link
        to={slug}
        key={slug}
        onMouseOver={() => this.setState({ showTitle: true })}
        onMouseLeave={() => this.setState({ showTitle: false })}
      >
        <div className="project">
          <Img fluid={featuredImage.fluid} className="project__thumbnail" />
          {this.state.showTitle ? (
            <span className="project__thumbnail__title">{title}</span>
          ) : null}
        </div>
      </Link>
    );
  }
}

ProjectThumbnail.propTypes = {
  project: PropTypes.shape({
    slug: PropTypes.string.isRequired,
    images: PropTypes.array.isRequired,
    title: PropTypes.string.isRequired,
  }).isRequired,
};

export default ProjectThumbnail;
