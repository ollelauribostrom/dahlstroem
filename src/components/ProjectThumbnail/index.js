import ProjectThumbnail from './ProjectThumbnail';
import './ProjectThumbnail.css';

export default ProjectThumbnail;
