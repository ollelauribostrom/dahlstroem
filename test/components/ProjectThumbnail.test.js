import React from 'react';
import { shallow } from 'enzyme';

import ProjectThumbnail from '../../src/components/ProjectThumbnail';

describe('Tests for ProjectThumbnail', () => {
  it('renders correctly', () => {
    const wrapper = shallow(
      <ProjectThumbnail
        hoverColor={{ hex: '#000' }}
        project={{
          slug: '/test',
          title: 'Test',
          featuredImage: {
            fluid: {
              aspectRatio: 1.4678899082568808,
              sizes: '(max-width: 1800px) 100vw, 1800px',
              src: 'someurl',
              srcSet: 'someurl',
            },
          },
        }}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
  it('shows the project title when hovering the thumbnail', () => {
    const wrapper = shallow(
      <ProjectThumbnail
        hoverColor={{ hex: '#000' }}
        project={{
          slug: '/test',
          title: 'Test',
          featuredImage: {
            fluid: {
              aspectRatio: 1.4678899082568808,
              sizes: '(max-width: 1800px) 100vw, 1800px',
              src: 'someurl',
              srcSet: 'someurl',
            },
          },
        }}
      />
    );
    expect(wrapper.state().showTitle).toEqual(false);
    wrapper.simulate('mouseover');
    expect(wrapper.state().showTitle).toEqual(true);
    expect(wrapper).toMatchSnapshot();
  });
});
