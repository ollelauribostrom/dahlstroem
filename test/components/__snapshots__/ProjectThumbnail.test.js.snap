// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for ProjectThumbnail renders correctly 1`] = `
<mockConstructor
  className="project__thumbnail"
  key="/test"
  onMouseLeave={[Function]}
  onMouseOver={[Function]}
  to="/test"
>
  <Image
    Tag="div"
    alt=""
    className="project__thumbnail__image"
    critical={false}
    fadeIn={true}
    fluid={
      Object {
        "aspectRatio": 1.4678899082568808,
        "sizes": "(max-width: 1800px) 100vw, 1800px",
        "src": "someurl",
        "srcSet": "someurl",
      }
    }
  />
</mockConstructor>
`;

exports[`Tests for ProjectThumbnail shows the project title when hovering the thumbnail 1`] = `
<mockConstructor
  className="project__thumbnail"
  key="/test"
  onMouseLeave={[Function]}
  onMouseOver={[Function]}
  to="/test"
>
  <Image
    Tag="div"
    alt=""
    className="project__thumbnail__image"
    critical={false}
    fadeIn={true}
    fluid={
      Object {
        "aspectRatio": 1.4678899082568808,
        "sizes": "(max-width: 1800px) 100vw, 1800px",
        "src": "someurl",
        "srcSet": "someurl",
      }
    }
  />
  <span
    className="project__thumbnail__title"
    style={
      Object {
        "backgroundColor": "#000",
      }
    }
  >
    Test
  </span>
</mockConstructor>
`;
